### PR TITLE
Axios mode

### PR DESCRIPTION
**Pull request by Eli-BB8**

### CODE REVIEW

File: src/util/diffsHandlerUtils/updatedDataHandler.js [hunk 1]
```diff
@@ -60,33 +57,67 @@ module.exports = async (diffsObj, dataSource, aka_all_data, currentUnit_to_DataS
             continue;
```
Comment (weizmm) [inline src/util/diffsHandlerUtils/updatedDataHandler.js:57]: change dataSource at line 55

File: src/util/diffsHandlerUtils/updatedDataHandler.js [hunk 2]
```diff
@@ -60,33 +57,67 @@ module.exports = async (diffsObj, dataSource, aka_all_data, currentUnit_to_DataS
...
         }
 
-        person = tryFindPerson.result;
+        DataModel.person = tryFindPerson.result;
 
         if (dataSource === fn.dataSources.aka) {
```
Comment (weizmm) [inline src/util/diffsHandlerUtils/updatedDataHandler.js:62]: dataSourcesssssssssssssssssssssss

File: src/util/diffsHandlerUtils/updatedDataHandler.js [hunk 3]
```diff
@@ -60,33 +57,67 @@ module.exports = async (diffsObj, dataSource, aka_all_data, currentUnit_to_DataS
...
-            updateSpecificFields(record[2], dataSource, person, record[1]);
+            updateSpecificFields(DataModel);
         } else {
-            let akaRecord = aka_all_data.find(person => ((person[fn.aka.personalNumber] == tryFindPerson.argument) || (person[fn.aka.identityCard] == tryFindPerson.argument)));
+            DataModel.akaRecord = aka_all_data.find(
+                person => (
```
Comment (weizmm) [inline src/util/diffsHandlerUtils/updatedDataHandler.js:66]: persom => record

File: src/util/diffsHandlerUtils/updatedDataHandler.js [hunk 4]
```diff
@@ -60,33 +57,67 @@ module.exports = async (diffsObj, dataSource, aka_all_data, currentUnit_to_DataS
...
+                    );
+                    
+                    const include = fn.akaRigid.includes(keyForCheck);
+                    if (include) {
+                        sendLog(
+                            logLevel.warn,
+                            logDetails.warn.WRN_AKA_FIELD_RIGID,
+                            diffsObj.path.toString(),
+                            tryFindPerson.argument,
+                            dataSource
```
Comment (weizmm) [inline src/util/diffsHandlerUtils/updatedDataHandler.js:108]: dataSorceeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

File: src/util/diffsHandlerUtils/updatedDataHandler.js [hunk 5]
```diff
@@ -60,33 +57,67 @@ module.exports = async (diffsObj, dataSource, aka_all_data, currentUnit_to_DataS
...
+                        )
+                    }
+
+                    return !include;
+                }
+            );
 
-            if (deepDiffForUpdate.length > 0) {
-                await updateSpecificFields(deepDiffForUpdate, dataSource, person, akaRecord, needMatchToKartoffel);
+            if (DataModel.deepDiffObj[2].length > 0) {
```
Comment (weizmm) [inline src/util/diffsHandlerUtils/updatedDataHandler.js:116]: updateDeepDiff

File: src/util/fieldsUtils/domainUserHandler.js [hunk 6]
```diff
@@ -13,45 +13,32 @@ const Auth = require('../../auth/auth');
  * @param {Object} originalRecord The original raw record before matchToKartoffel 
  *
  *  */
-module.exports = async (person, record, dataSource, needMatchToKartoffel=true, originalRecord) => {
+// module.exports = async (person, record, dataSource, needMatchToKartoffel = true, originalRecord) => {
```
Comment (weizmm) [inline src/util/fieldsUtils/domainUserHandler.js:16]: remove comment

File: src/util/getIdentifiers.js [hunk 7]
```diff
@@ -12,22 +12,15 @@ const fn = require('../config/fieldNames');
...
-
+const getIdentifiers = async (record, dataSource) => {
     let recordRelevants = {};
 
     fn[dataSource].idsFields.forEach(field => {
         recordRelevants[fn[dataSource][field]] = record[fn[dataSource][field]];
     });
 
     let ids = await matchToKartoffel(recordRelevants, dataSource);
+
```
Comment (weizmm) [inline src/util/getIdentifiers.js:23]: return according to entityType

File: src/util/recordsFilter.js [hunk 8]
```diff
@@ -6,15 +6,15 @@ const filterAka = require('./filters/aka/filterAka');
 /**
  * Filter records by checking their data
  *
- * @param {Array<Object>} records - array of raw data about each person
+ * @param {Array<DataModel>} dataModels - array of raw data about each person
  * @param {string} dataSource - the name of the data source
  */
-module.exports = async (records, dataSource, flowType) => {
+module.exports = async (dataModels, dataSource) => {
```
Comment (weizmm) [inline src/util/recordsFilter.js:12]: dataSourceeeeeeeeeeeeeeeeeee

File: src/util/DataModel.js [hunk 9]
```diff
@@ -1,20 +1,20 @@
...
+    constructor(record, dataSource, flowType, runningType, updateDeepDiff) {
         this.record = record;
-        this.deepDiffObj = deepDiffObj;
+        this.updateDeepDiff = updateDeepDiff;
         this.dataSource = dataSource;
         this.flowType = flowType;
         this.runningType = runningType;
+        this.identifiers = [];
         this.needMatchToKartoffel = true;
+        this.needCompleteFromAka = true;
```
Comment (Eli-BB8) [inline src/util/DataModel.js:13]: semantic change remove the need

File: src/util/diffsHandlerUtils/addedDataHandler.js [hunk 10]
```diff
@@ -17,12 +17,12 @@ require('dotenv').config();
...
  *
- * @param { DataModel[] } newData - represnts the changes from last data
+ * @param { DataModel[] } addedData - represnts the changes from last data
  * @param {*} aka_all_data - all the data from aka data source (for compilation)
  * @param {*} currentUnit_to_DataSource - a map of all units from each data source
  */
-module.exports = async (newData, dataSource, aka_all_data, currentUnit_to_DataSource) => {
-    let dataModels = await recordsFilter(newData, dataSource);
+module.exports = async (addedData, aka_all_data, currentUnit_to_DataSource) => {
+    let dataModels = await recordsFilter(addedData, addedData[0].dataSource);
```
Comment (Eli-BB8) [inline src/util/diffsHandlerUtils/addedDataHandler.js:25]: dont send dataSource && change the filter

File: src/util/diffsHandlerUtils/addedDataHandler.js [hunk 11]
```diff
@@ -77,7 +77,14 @@ module.exports = async (newData, dataSource, aka_all_data, currentUnit_to_DataSo
                 // Add the complete person object to Kartoffel
                 try {
                     if (!DataModel.person_ready_for_kartoffel.directGroup) {
-                        // log is neccarry
+                        sendLog(
```
Comment (Eli-BB8) [inline src/util/diffsHandlerUtils/addedDataHandler.js:80]: new warnnig log

File: src/util/diffsHandlerUtils/addedDataHandler.js [hunk 12]
```diff
@@ -114,39 +121,38 @@ module.exports = async (newData, dataSource, aka_all_data, currentUnit_to_DataSo
...
+                    DataModel.dataSource,
                     errMessage
                 );
             }
         } else if (tryFindPerson.result) {
 
             DataModel.person = tryFindPerson.result;
 
-            DataModel.isDataSourcePrimary = (currentUnit_to_DataSource.get(DataModel.person_ready_for_kartoffel.currentUnit) === dataSource);
+            DataModel.checkIfDataSourceIsPrimary(currentUnit_to_DataSource);
```
Comment (Eli-BB8) [inline src/util/diffsHandlerUtils/addedDataHandler.js:132]: check if possible to require the map in the dataModel

File: src/util/diffsHandlerUtils/updatedDataHandler.js [hunk 13]
```diff
@@ -20,10 +20,10 @@ require('dotenv').config();
  * @param {Map} currentUnit_to_DataSource - map of all the units from each data source
  * @param {boolean} needMatchToKartoffel - if the diffsObj needs match to kartoffel
  */
-module.exports = async (updatedData, dataSource, aka_all_data, currentUnit_to_DataSource) => {
+module.exports = async (updatedData, aka_all_data, currentUnit_to_DataSource) => {
     let dataModels = updatedData;
 
-    dataModels = await recordsFilter(dataModels, dataSource);
+    dataModels = await recordsFilter(dataModels, dataModels[0].dataSource);
```
Comment (Eli-BB8) [inline src/util/diffsHandlerUtils/updatedDataHandler.js:26]: remove data source and change filter

File: src/util/getIdentifiers.js [hunk 14]
```diff
@@ -13,26 +13,26 @@ const fn = require('../config/fieldNames');
...
 
-    fn[dataSource].idsFields.forEach(field => {
-        recordRelevants[fn[dataSource][field]] = record[fn[dataSource][field]];
-    });
-
-    let ids = await matchToKartoffel(recordRelevants, dataSource);
+    if (matchedRecord.entityType === fn.entityTypeValue.s || matchedRecord.entityType === fn.entityTypeValue.c) {
+        return matchedRecord;
+    } else if (matchedRecord.entityType === fn.entityTypeValue.gu) {
+        (dataSource === fn.dataSources.ads && record[fn[dataSource].sAMAccountName]) ?
```
Comment (Eli-BB8) [inline src/util/getIdentifiers.js:22]: extarct to function and change also in domainUserHandler

**Review by weizmm — APPROVED** [review]

**Pull request merged**